pkg/asset/installconfig/platform.go

```diff
@@ -131,6 +149,29 @@ func (a *platform) queryUserForPlatform() (platform string, err error) {
 	return
 }
 
+func (a *platform) queryUserForArchitecture(archlist []string) (architecture types.Architecture, err error) {
```

Comment:
I don't think that this meets the bar for inclusion in the survey. The user can manually set the architecture in the install-config.yaml.

Reply:
so ARM will be a supported architecture from hereon for AWS, so don't we need that option for the user to select the architecture? I'm perfectly fine manually changing the architecture, but trying to understand the reason why we would not want this.

Reply:
Manual change should not be required.  If we could set the AWS architecture based on the payload architecture, that might suffice in place of a question, but not sure how to go about that.

Reply:
> so ARM will be a supported architecture from hereon for AWS, so don't we need that option for the user to select the architecture? I'm perfectly fine manually changing the architecture, but trying to understand the reason why we would not want this.

The set of survey questions is not meant to be exhaustive of all of the configuration that a user may do. The set is meant to be small and curated to only configuration that the installer cannot provide opinionated defaults for. In this case, the installer can choose a default architecture.

Reply:
Not sure that choosing an architecture is a configuration option, it's a fundamental decision regarding where and on what to host a cluster, since x86 and ARM clusters aren't binary compatible.  It also affects the decision about regions, which is already a question.
From the user experience perspective, the results of `create install-config` "just work" without modification for deploying a default cluster.  ARM users should have the same experience.

Reply:
The question to ask is "Can I install a cluster without the user providing this configuration?"
* SSH key - yes, but in older versions the answer was no, so this has been grandfathered in
* Platform - no
* Region - arguably yes, but the user may not have access to whatever the default region would be in their account
* Base domain - no
* Cluster name - no
* Pull secret - no

The answer for architecture is yes, so does not meet the requirements for inclusion. The criteria is not whether configuring through the survey makes the experience better for the user that wants that configuration.

What percentage of users do you expect would choose ARM over x86?

Reply:
Payloads are architecture specific.  Therefore, if the installer just defaults to amd64, then the answer to your question will actually be "no", because the installation will fail with a non-x86 payload without manual modification of the configuration.  Furthermore, even if the user were to adjust the architecture, if they selected a region in which ARM instances aren't available, the installation will fail in that case too.

Reply:
> Payloads are architecture specific. Therefore, if the installer just defaults to amd64, then the answer to your question will actually be "no", because the installation will fail with a non-x86 payload without manual modification of the configuration.
Then the user does not have any choice in this, and asking the user to make a selection is wrong. The installer needs to make this decision for the user.

> Furthermore, even if the user were to adjust the architecture, if they selected a region in which ARM instances aren't available, the installation will fail in that case too.

I don't follow how that is relevant here. If the user updates the install-config.yaml to select the ARM architecture, then they are obligated to select an appropriate region as well. If they do not, then the installer should rightly fail validation.

Reply:
Is there a way for the installer to know the payload architecture during `create install-config`?

Reply:
It would need to be built into the installer binary.

Reply:
should we pursue this discussion in a separate issue and remove the addition of the install-config option here so we can proceed with this PR? this discussion encompasses more than just ARM and includes all other architectures in general.

Reply:
https://github.com/openshift/installer/pull/4870#issuecomment-828909878